test/VocabularyTest.cpp

```diff
@@ -94,6 +109,36 @@ TEST(VocabularyTest, createFromSetTest) {
   ASSERT_FALSE(v.getId("foo", &id));
 };
 
+// ______________________________________________________________________________________________
+TEST(VocabularyTest, StringSortComparator) {
+  StringSortComparator comp(true);
+
+  // strange casings must not affect order
+  ASSERT_TRUE(comp("ALPHA", "beta"));
+  ASSERT_TRUE(comp("alpha", "BETA"));
+  ASSERT_TRUE(comp("AlPha", "bEtA"));
+  ASSERT_TRUE(comp("AlP", "alPha"));
+  ASSERT_TRUE(comp("alP", "ALPha"));
+
+  // inverse tests for completeness
+  ASSERT_FALSE(comp("beta", "ALPHA"));
+  ASSERT_FALSE(comp("BETA", "alpha"));
+  ASSERT_FALSE(comp("bEtA", "AlPha"));
+  ASSERT_FALSE(comp("alPha", "AlP"));
+  ASSERT_FALSE(comp("ALPha", "alP"));
+
+  // only if lowercased version is exactly the same we want to sort by the
+  // casing (upper-case letters have lower ascii values than lower-case)
+  ASSERT_FALSE(comp("alpha", "ALPHA"));
+  ASSERT_TRUE(comp("ALPHA", "alpha"));
+
+  // TODO: check what to do about these cases
+  // ASSERT_TRUE(comp("\"Hannibal\"@en", "\"Hannibal Hamlin\"@en"));
```

Comment:
Shouldn't this work with the latest commit that splits of the language tag and only compares values?